test/NFTOpt.tests.ts

```diff
@@ -58,6 +84,134 @@ describe("NFTOpt Tests", function () {
         it("should test that method can be called", async function () {
             expect(NFTOptCTR.connect(buyer).createOption(0)).to.not.throw;
         });
+
+        it("should fail when the option with the specified id does not exist", async function () {
+            await expect(
+                NFTOptCTR.connect(seller).createOption(0)
+              ).to.be.revertedWith("Option with the specified id does not exist");
+        });
+
+        it("should fail when the option is already fulfilled by a seller", async function () {
+            await NFTOptCTR.connect(buyer).publishOptionRequest(
+                dummyOptionRequest.nftContract,
+                dummyOptionRequest.nftId,
+                dummyOptionRequest.strikePrice,
+                dummyOptionRequest.interval,
+                dummyOptionRequest.flavor,
+                {value: dummyOptionRequest.premium}
+            )
+
+            await NFTOptCTR.connect(seller).createOption(1);
+
+            await expect(
+                NFTOptCTR.connect(seller).createOption(1)
+              ).to.be.revertedWith("Option is already fulfilled by a seller");
+
+        });
+
+        it("should fail when the option is not in the request state", async function () {
+            await NFTOptCTR.connect(buyer).publishOptionRequest(
```

Comment:
wrap with expect ().to.not.be.reverted

Reply:
also, I think it would be better to extract this code into a helper method and call that everywhere this is called; 

could have the method named `publishOptionRequestDummy`